File: go/blink.go
package main

import (
  "context"
  "fmt"
  "time"

  "github.com/edaniels/golog"
  "go.viam.com/rdk/robot/client"
  "go.viam.com/rdk/utils"
  "go.viam.com/utils/rpc"
  "go.viam.com/rdk/components/board"
)

func main() {
  logger := golog.NewDevelopmentLogger("client")
  robot, err := client.New(
      context.Background(),
      "[ADD YOUR ROBOT ADDRESS HERE. YOU CAN FIND THIS ON THE CONNECT TAB OF THE VIAM APP]",
      logger,
      client.WithDialOptions(rpc.WithCredentials(rpc.Credentials{
          Type:    utils.CredentialsTypeRobotLocationSecret,
          Payload: "[PLEASE ADD YOUR SECRET HERE. YOU CAN FIND THIS ON THE CONNECT TAB OF THE VIAM APP]",
      })),
  )
  if err != nil {
      logger.Fatal(err)
  }
  defer robot.Close(context.Background())
  logger.Info("Resources:")
  logger.Info(robot.ResourceNames())

  myBoard, err := board.FromRobot(robot, "myBoard")
  if err != nil {
    logger.Fatalf("could not get board: %v", err)
  }

  led, err := myBoard.GPIOPinByName("8")
  if err != nil {
    logger.Fatalf("could not get led: %v", err)
  }

  for {
    err = led.Set(context.Background(), true, nil)
    if err != nil {
      logger.Fatalf("could not set led to on: %v", err)
    }
    fmt.Println("LED is on")

    time.Sleep(1 * time.Second)
    err = led.Set(context.Background(), false, nil)
    if err != nil {
      logger.Fatalf("could not set led to off: %v", err)
    }
    fmt.Println("LED is off")
    time.Sleep(1 * time.Second)
  }

}

File: python/blink.py
import asyncio

from viam.robot.client import RobotClient
from viam.rpc.dial import Credentials, DialOptions
from viam.components.board import Board


async def connect():
    async def connect():
        opts = RobotClient.Options.with_api_key(
            # Replace "<API-KEY>" (including brackets) with your machine's API key
            api_key='<API-KEY>',
            # Replace "<API-KEY-ID>" (including brackets) with your machine's
            # API key ID
            api_key_id='<API-KEY-ID>'
        )
        return await RobotClient.at_address("ADDRESS FROM THE VIAM APP", opts)

async def main():
    robot = await connect()

    print('Resources:')
    print(robot.resource_names)

    local = Board.from_robot(robot, 'local')
    led = await local.gpio_pin_by_name('8')

    while (True):
        # When True, sets the LED pin to high or on.
        await led.set(True)
        print('LED is on')
        await asyncio.sleep(1)

        # When False, sets the pin to low or off.
        await led.set(False)
        print('LED is off')
        await asyncio.sleep(1)

    await robot.close()

if __name__ == '__main__':
    asyncio.run(main())
